Add an article about decentralized storage solutions

bucanero opened the pull request:
Fixes https://github.com/near/docs/issues/683

BenKurrek commented on docs/concepts/storage.md:
```diff
@@ -0,0 +1,89 @@
+
+### Content identifier
+
+When you add a file to IPFS, your file is split into smaller chunks, cryptographically hashed, and given a unique fingerprint called a content identifier (CID). 
```
```suggestion
When you add a file to IPFS, it is split into smaller chunks, cryptographically hashed, and given a unique fingerprint called a content identifier (CID). 
```

BenKurrek commented on docs/concepts/storage.md:
```diff
@@ -0,0 +1,89 @@
+IPFS offers a decentralized naming system so you don't need to remember a long string of CIDs.
+IPFS can find the latest version of your file using the IPNS decentralized naming system, and you can use DNSLink to map CIDs to human-readable DNS names.
+
+### IPFS providers
```
Is it possible to make all these headers one size down? It's kinda hard to notice the difference between the main sections (SIA, IPFS, ARWEAVE), and their subheadings as shown here: 
![image](https://user-images.githubusercontent.com/57506486/137491066-07272579-832a-4848-973f-f377e466f075.png)

BenKurrek commented on docs/concepts/storage.md:
```diff
@@ -0,0 +1,89 @@
+
+### Privacy
+
+Sia encrypts and distributes your files across a decentralized network. With Sia you control the private encryption keys and you own the data, so no third-party can access or control your files.
```
```suggestion
Sia encrypts and distributes your files across a decentralized network. With Sia, you control the private encryption keys and you own the data, so no third-party can access or control your files.
```

thisisjoshford commented on website/sidebars.json:
```diff
@@ -9,7 +9,8 @@
       "concepts/networks",
-      "concepts/transaction"
+      "concepts/transaction",
+      "concepts/storage-solutions"
```
This looks a little confusing in the sidebar as we have multiple sections that talk about storage. As a new user, I think I would be a little lost when trying to navigate this. Let's have a sub-menu for "Data Storage" that has two parts. "Storage on NEAR" and "Alternative Solutions" or something like this.

bucanero replied:
I've grouped the articles together, under a "Data Storage" category:
- Data Storage
   - Storage on NEAR
   - Storage Staking
   - Alternative Solutions

thisisjoshford reviewed (changes requested):
Great overview of these three solutions. I think we should add simple examples of their implementation. Perhaps @doriancrutcher could assist with this?

bucanero commented:
> Great overview of these three solutions. I think we should add simple examples of their implementation. Perhaps @doriancrutcher could assist with this?

hi @thisisjoshford, I have updated the draft and added examples for each storage (thanks to @doriancrutcher !!)

let me know what you think 😃
